### ISSUE TITLE
Conan install on Windows (Visual Studio 17 2022)

### ISSUE THREAD

**Issue by AndreaLanfranchi** [symbol/symbol#1012]
Trying to follow build instructions for conan install on Windows/Visual Studio.
The step to build missing dependencies with conan aborts on error
```
[...]
mongo-c-driver/1.25.4@nemtech/stable: Already installed!
openssl/3.2.1: Already installed!
rocksdb/8.9.1@nemtech/stable: Already installed!
zeromq/4.3.5@nemtech/stable: Already installed!
cppzmq/4.10.0@nemtech/stable: Already installed!
mongo-cxx-driver/3.9.0@nemtech/stable: Already installed!
conanfile.py: Applying build-requirement: gtest/1.14.0
conanfile.py: Applying build-requirement: benchmark/1.8.3@nemtech/stable
conanfile.py: Generator txt created conanbuildinfo.txt
conanfile.py: Calling generate()
ERROR: conanfile.py: Error in generate() method, line 87
        if dep.cpp_info.libdirs and require.libs and require.run:
        AttributeError: 'Requirement' object has no attribute 'libs'
```

Opening conanfile.py and inserting a `print(dir(require))` (roughly at line 84) reveals this
```
['__class__', '__delattr__', '__dict__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__getstate__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__le__', '__lt__', '__module__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', '__weakref__', 'aggregate', 'build', 'direct', 'ref', 'test', 'visible']
```

So apparently no "libs" nor "run" attributes.
Any hint about how to solve ?

**Comment by AndreaLanfranchi** [comment]
P.s. I'm referring to the build of client/catapult

**Comment by Wayonb** [comment]
What version of Conan are you using?

**Comment by AndreaLanfranchi** [comment]
Name: conan
Version: 1.58.0
Summary: Conan C/C++ package manager
Home-page: https://conan.io
Author: JFrog LTD
Author-email: luism@jfrog.com
License: MIT
Location: C:\Users\Andrea\AppData\Roaming\Python\Python311\site-packages
Requires: bottle, colorama, fasteners, Jinja2, node-semver, patch-ng, pluginbase, pygments, PyJWT, python-dateutil, PyYAML, requests, six, tqdm, urllib3

**Comment by AndreaLanfranchi** [comment]
I realize now I'm probably on outdated version of Conan.
Strangely enough I have python 3.11 installed and `pip install conan` installed version 1.58 instead of more recent 2.x

**Comment by AndreaLanfranchi** [comment]
Solved by upgrading to conan 2.2
Thank you.

P.s. Probably a check for conan version in conanfile.py should be enforced.
